dom.go

```diff
@@ -1612,6 +1661,14 @@ type BasicElement struct {
 	*BasicNode
 }
 
+func (e *BasicElement) Attributes() []Attribute {
+	return attributeListToAttributes(e.Get("attributes"))
+}
+
+func (e *BasicElement) Dataset() []Attribute {
+	return dataListToAttributes(e.Get("attributes"))
```

Comment:
Generally speaking, `js/dom` is a rather low-level library. Unlike something like jQuery, it doesn't try to work around flaws, bugs or old versions. As such, it would be preferred to use `dataset` directly, and break if it's not available.

Optimally, people would use a library on top of `js/dom` that acts like jQuery, providing a unified API that works around all the issues, differences and lacking features.